nixos/modules/services/misc/bcg.nix

```diff
@@ -159,7 +159,7 @@ in
     in {
       description = "BigClown Gateway";
       wantedBy = [ "multi-user.target" ];
-      wants = mkIf config.services.mosquitto.enable [ "mosquitto.service" ];
+      wants = mkMerge [ [ "network-online.target" ] (mkIf config.services.mosquitto.enable [ "mosquitto.service" ]) ];
```

Comment:
Can't this just be a `["network-online.target"] + lib.optional config.services.mosquitto.enable "mosquitto.service";`?